# <center>Escuela Militar de Ingeniería</center>
<center>Ingeniería de Sistemas</center>
<center><img src="https://i.servimg.com/u/f73/20/03/60/99/logo_i12.jpg" width="300"></center>
<center>Estudiante: Zalles Merino Izai Alejandro</center>
<center>Código SAGA: A27006-7</center>

## <center>Docente: Ing. Angela Santos</center>
## Fecha: 11 de Abril de 2025

In [ ]:
# Instalar paquetes necesarios
if (!require("knitr")) install.packages("knitr", quiet = TRUE)
library(knitr)

# Simulación Monte Carlo - Semillas

In [ ]:
# Paso 1: Tabla base de frecuencias y probabilidades
categorias <- c("Buena", "Baja", "NoDesarrollada")
frecuenciaAbs <- c(5, 3, 2)
frecuenciaRel <- frecuenciaAbs / sum(frecuenciaAbs)
frecuenciaAcum <- cumsum(frecuenciaRel)
costos <- c(7, 3, 0)

In [ ]:
# Intervalos acumulados
intervaloInferior <- c(0, head(frecuenciaAcum, -1))
intervaloSuperior <- frecuenciaAcum
tablaDistribucion <- data.frame(
  Categoria = categorias,
  FrecuenciaAbs = frecuenciaAbs,
  FrecuenciaRel = round(frecuenciaRel, 2),
  FrecuenciaAcum = round(frecuenciaAcum, 2),
  CostoUnidad = costos,
  Intervalo = paste0(sprintf("%.2f", intervaloInferior), " - ", sprintf("%.2f", intervaloSuperior - 0.01))
)
cat("🟦 Tabla 1: Distribución de categorías\n")
kable(tablaDistribucion)

🟦 Tabla 1: Distribución de categorías




|Categoria      | FrecuenciaAbs| FrecuenciaRel| FrecuenciaAcum| CostoUnidad|Intervalo   |
|:--------------|-------------:|-------------:|--------------:|-----------:|:-----------|
|Buena          |             5|           0.5|            0.5|           7|0.00 - 0.49 |
|Baja           |             3|           0.3|            0.8|           3|0.50 - 0.79 |
|NoDesarrollada |             2|           0.2|            1.0|           0|0.80 - 0.99 |

In [ ]:
# Paso 2: Simulación Monte Carlo (100 semillas)
# Quitamos set.seed(123) para que los números sean realmente aleatorios
numSemillas <- 100
numerosAleatorios <- runif(numSemillas)

In [ ]:
# Corrección en la función de clasificación para usar los intervalos correctos
ClasificarCategoria <- function(x) {
  if (x < frecuenciaAcum[1]) {
    return(categorias[1])  # Buena
  } else if (x < frecuenciaAcum[2]) {
    return(categorias[2])  # Baja
  } else {
    return(categorias[3])  # NoDesarrollada
  }
}
categoriasSimuladas <- sapply(numerosAleatorios, ClasificarCategoria)

In [ ]:
# Codificación One-Hot
oneHot <- data.frame(
  Buena = as.integer(categoriasSimuladas == "Buena"),
  Baja = as.integer(categoriasSimuladas == "Baja"),
  NoDesarrollada = as.integer(categoriasSimuladas == "NoDesarrollada")
)
tablaSimulacion <- data.frame(
  Semilla = 1:numSemillas,
  Aleatorio = round(numerosAleatorios, 4),
  Categoria = categoriasSimuladas,
  oneHot
)
cat("\n🟦 Tabla 2: Simulación Monte Carlo (Todas las semillas)\n")
kable(tablaSimulacion)  # Mostramos todas las semillas


🟦 Tabla 2: Simulación Monte Carlo (Todas las semillas)




| Semilla| Aleatorio|Categoria      | Buena| Baja| NoDesarrollada|
|-------:|---------:|:--------------|-----:|----:|--------------:|
|       1|    0.3536|Buena          |     1|    0|              0|
|       2|    0.3664|Buena          |     1|    0|              0|
|       3|    0.2871|Buena          |     1|    0|              0|
|       4|    0.0800|Buena          |     1|    0|              0|
|       5|    0.3655|Buena          |     1|    0|              0|
|       6|    0.1780|Buena          |     1|    0|              0|
|       7|    0.5361|Baja           |     0|    1|              0|
|       8|    0.5039|Baja           |     0|    1|              0|
|       9|    0.9450|NoDesarrollada |     0|    0|              1|
|      10|    0.3413|Buena          |     1|    0|              0|
|      11|    0.4647|Buena          |     1|    0|              0|
|      12|    0.0825|Buena          |     1|    0|              0|
|      13|    0.8601|NoDesarrollada |     0|    0|          

In [ ]:
# Paso 3: Resultados finales
resumenCategorias <- c(
  sum(oneHot$Buena),
  sum(oneHot$Baja),
  sum(oneHot$NoDesarrollada)
)
gananciaTotal <- resumenCategorias[1] * costos[1] +
                 resumenCategorias[2] * costos[2] +
                 resumenCategorias[3] * costos[3]

tablaResumen <- data.frame(
  Categoria = categorias,
  Conteo = resumenCategorias,
  Ganancia = c(
    resumenCategorias[1] * costos[1],
    resumenCategorias[2] * costos[2],
    resumenCategorias[3] * costos[3]
  )
)
cat("\n🟦 Tabla 3: Resumen de resultados\n")
kable(tablaResumen)
cat("\n💰 Ganancia total:", gananciaTotal, "Bs.\n")


🟦 Tabla 3: Resumen de resultados




|Categoria      | Conteo| Ganancia|
|:--------------|------:|--------:|
|Buena          |     45|      315|
|Baja           |     28|       84|
|NoDesarrollada |     27|        0|


💰 Ganancia total: 399 Bs.


Los resultados muestran que la categoría "Buena" tiene el mayor número de semillas siendo 45 y genera la mayor ganancia de 315 Bs. Le sigue la categoría "Baja" con 28 semillas y una ganancia de 84 Bs. Por último, la categoría "No Desarrollada" tiene 27 semillas, pero no genera ganancias. En total, la ganancia alcanzada es de 399 Bs., lo que indica que las categorías "Buena" y "Baja" contribuyen positivamente a las ganancias, mientras que "No Desarrollada" no aporta.